### PR TITLE
Improve network selection flow

### PR DESCRIPTION
**Pull request by justindg**
This PR includes a rework the network selection flow as discussed in #1821 and #1804. 

Summary of changes:
- Removed references to `getDefaultNetwork()` and `setDefaultNetworkInfo()` in all classes not related to DappBrowser. Also renamed them to `getActiveBrowserNetwork()` and `setActiveBrowserNetwork()` respectively. 
- Network changes in `MyAddressActivity` (POS mode) will only reflect locally
- Cleaned up unused methods related to defaultNetwork
- Updated Testnet Modal UI
- Removed unused class `GetDefaultWalletBalance`. Note: `FindDefaultNetworkInteract` can also be removed by refactoring a few files, as the it only contains duplicate methods found in EthereumNetworkRepository. (let me know if you wish to do this)
- Removed references to `isActiveMainNet()` and `setActiveMainNet()`, storing these in SharedPrefs causes errors in the new network selection interaction. Value of `isActiveMainNet` can be derived from the stored chain ids.
- Single selection mode will not affect stored filterNetworks
- Once active browser network is deselected via Settings, `activeBrowserNetwork` will be set to `null`. Whenever DappBrowser becomes visible to user, the app will search for network changes. A null value will prompt the user to connect to a network.
- Prevent the user from exiting network selection without selecting at least one value. For now, a toast message will be displayed.

### CODE REVIEW

**Review by JamesSmartCell — CHANGES_REQUESTED** [review]
Sorry about the confusion, my mistake I should drink coffee in the morning before looking at PRs!

So, in summary all that needs changing is:

1. when you exit from the SelectNetworkActivity, call the RESET_WALLET message as previously was in there - otherwise, you may still be showing testnet/mainnet  tokens after the switch.
2. No need to use 'hasRealValue' and 'isTestNet', as they are always boolean opposite.

I think that's it.

**Comment by justindg** [comment]
Hi @JamesSmartCell, regarding (1), I intended to display all mainnets/testnets in single selection mode because I saw this comment from @colourfreak in the original thread: https://github.com/AlphaWallet/alpha-wallet-android/issues/1804#issuecomment-824068805, also in [Zeplin](https://app.zeplin.io/project/5d088205bff2d15de6a4397b/screen/608026cdd784971b6b2c0b01). Can you please confirm this?

Regarding (2) - this can easily be done however I don't see this particular interaction in Zeplin

File: app/src/main/java/com/alphawallet/app/repository/EthereumNetworkBase.java [hunk 1]
```diff
@@ -264,40 +265,27 @@ public NetworkInfo getNetworkByChain(int chainId)
     public List<Integer> getFilterNetworkList()
     {
         String filterList = preferences.getNetworkFilterList();
-        boolean mainNets = preferences.isActiveMainnet();
         List<Integer> storedIds = Utils.intListToArray(filterList);
         List<Integer> networkIds = new ArrayList<>();
```
[the remark this answers is not in the document]
Comment (justindg) [reply app/src/main/java/com/alphawallet/app/repository/EthereumNetworkBase.java]: Resolved

**Comment by JamesSmartCell** [comment]
Thankyou @justindg. It now works perfectly and code is cleaner than when you started.

**Review by JamesSmartCell — APPROVED** [review]

**Pull request merged**